airflow/kubernetes/pod_launcher.py

```diff
@@ -63,7 +63,16 @@ def __init__(self,
 
     def run_pod_async(self, pod, **kwargs):
         """Runs POD asynchronously"""
-        pod_mutation_hook(pod)
+        import airflow.configuration as conf
+        if conf.get("kubernetes", "pod_mutation_using_k8s_pod"):
+            pod_mutation_hook(pod)
+        else:
+            from airflow.kubernetes.pod import Pod
+            from airflow.kubernetes.pod_generator import PodGenerator
+            dummy_pod = Pod(image="", envs={}, cmds=[])
+            pod_mutation_hook(dummy_pod)
+            dummy_pod = dummy_pod.to_v1_kubernetes_pod()
+            PodGenerator.reconcile_pods(pod, dummy_pod)
```

Comment:
Same comment as above:

I think we will need to support both for 1.10.12 since some of the users have already updated their `pod_mutation_hook` to use k8s.V1Pod.

We could just check the type of the Pod that is passed, example:


```
if isinstance(k8s.V1Pod):
    # Use Pod from k8s api
    ....
else:
    # Use Pod from Airflow POD
```

Reply:
The change in PR won't work if we have the following in our ``airflow_local_settings.py`` (We have this for Astronomer- Airflow chart) since the pod labels would show up as `{}`:

```python
    def pod_mutation_hook(pod: Pod):
        extra_labels = {
            "kubernetes_executor": "False",
            "kubernetes_pod_operator": "False"
        }
        if 'airflow-worker' in pod.labels.keys() or \
                conf.get('core', 'EXECUTOR') == "KubernetesExecutor":
            extra_labels["kubernetes_executor"] = "True"
        else:
            extra_labels["kubernetes_pod_operator"] = "True"
```